assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -44,13 +49,48 @@ interactivityApiStore( {
 					context?.woocommerce.imageId
 				);
 			},
+			pagerDotFillOpacity( store: SelectorsStore ) {
+				const { context } = store;
+
+				return context?.woocommerce.selectedImage ===
+					context?.woocommerce.imageId
+					? 1
+					: 0.2;
+			},
 		},
 	},
 	actions: {
 		woocommerce: {
 			handleClick: ( { context }: Store ) => {
 				context.woocommerce.selectedImage = context.woocommerce.imageId;
 			},
+			onNextImageButtonClick: ( store: Store ) => {
```

Comment:
Usually, the handler function should have the prefix `handle`. What do you think?

```suggestion
			handleNextImageButtonClick: ( store: Store ) => {
```

Also, we should decide if we want a nested structure or not (https://github.com/woocommerce/woocommerce-blocks/pull/10823#discussion_r1325598086). Mmm, I don't have a strong preference, but it makes sense if we have a common approach.

Reply:
I just changed it from `onNext/onPrevious` to `handleNext/handlePrevious` so we establish a common pattern to follow

Regarding the nested structure, I believe that since the store is for the entire block (the Product Gallery block), we can leave it flat. However, if there are actions, selectors, or state that should only be used in the context of one of the inner blocks, we can nest them (this way, we make it clear that they should only be used within that specific inner block).